src/main/java/edu/pw/eiti/pik/user/UserController.java

```diff
@@ -47,6 +54,25 @@ UserDto me() {
         return userMapper.toDto(user);
     }
 
+    @ApiOperation(value = "Get top users by name who have given role (up to number)")
+    @ApiResponses({
+    	@ApiResponse(code = 200, message = "Always")
+    })
+    @GetMapping(path = "/user/{authority}/{name}/{number}")
+    List<UserDto> getUsersByRoleAndName(
+    		@PathVariable String authority,
+    		@PathVariable String name,
+    		@PathVariable int number)
+    {
+    	Page<User> queryResult = userService.findByNameAndAuthorityName(
+    			name.toLowerCase(), authority, PageRequest.of(0, number));
+	    ArrayList<UserDto> listToReturn = new ArrayList<>(); 
```

Comment:
I think you don't have to split query result and list to return. Just use streams.

`return userService.findByNameAndAuthorityName(
    			name.toLowerCase(), authority, PageRequest.of(0, number))
                .stream()
                .map(userMapper::toDto)
                .collect(Collectors.toList());`